### ISSUE TITLE
Display chart templates

### ISSUE THREAD

**Issue by cpoole** [helm/monocular#612]
It would be really nice to display the chart templates in a tab or second page under the chart page. 

I find myself mostly just using github to view charts because reading the templates is usually required to really figure out what is happening. 

I'd be happy to add an untar and save the templates into mongo if that would be accepted.

**Comment by prydonius** [comment]
We used to have a way to point to the source on GitHub, but this is difficult to do with more repositories as they don't all follow the same structure as the stable and incubator repositories. I definitely agree that this was a useful feature, it's very helpful to easily be able to dig into the chart templates from the Hub.

Storing this in MongoDB on sync and serving this could be good, though I'd like to understand what the implications are of that (e.g. how much more storage space might be expected), and how we should represent this data in the database. Do you think you'd be able to put together a proposal on this?

**Comment by cpoole** [comment]
Yep, that was my thought. Adding this to the sync job to untar the charts and save to mongo sounds the easiest. 

I can put together a formal proposal this weekend, but some quick napkin math says 

As an upper bound we can assume all charts are 1MB gzipped (configmap limit) + 40% for uncompressed storage.

There are 543 charts right now on helm hub, so keeping just the current version (or latest n versions) untarred "hot" in mongo seems rather trivial. I'd bet it's no more than 1G to keep the latest 3. 

As for charts older than the latest 3... I think v1 would just not show them, and maybe v2 could do an "on click" fetch and untar? That has DDoS potential though :thinking: 

If this sounds like the right track I'll go ahead and formalize this over the next few days.

**Comment by prydonius** [comment]
@cpoole thanks for the quick thoughts, that definitely doesn't sound too concerning. I didn't consider storing templates for previous chart versions. It would be interesting to be able to see changes between versions. For simplicity, we may just want to start with just storing the templates from the latest version, but keep historical versions in mind when designing the data structure.

Look forward to seeing the full proposal!
